frontend/src/contexts/UploadContext.js

```diff
@@ -105,9 +98,28 @@ export default function UploadContextProvider(props) {
         uploadLoadedFiles,
         allFiles,
         setPersistenceScope,
+        invalidateCache,
       }}
     >
       {props.children}
     </UploadContext.Provider>
   )
 }
+
+/**
+ * Fetch the files in the gitea repo associated with project being updated
+ * @param repo{string}
+ * @param shouldFetch{boolean}
+ * @returns {{isLoading: boolean, isError: boolean, files: any | Array | null}}
+ */
+const useFetchRemoteRepoFiles = (repo, shouldFetch) => {
+  const fetcher = () => getDefaultBranchFiles(repo)
+
+  const { data, error } = useSWR(shouldFetch ? `update/${repo}` : null, fetcher)
```

Comment:
This is a really weird one.

Reply:
It's exactly as the [docs](https://swr.vercel.app/docs/conditional-fetching) say how it should be done :sweat_smile:

Reply:
Oh, haha, hadn't seen that. Could still use a comment I think. Also, what's going on here, why do we call `useFetchRemoteRepoFiles` when we don't actually want the files?

The rest of what I said about the keys still stands anyway.

Reply:
The value of `shouldFetch` is true if the current page is under the `projects/update` route. But `UploadContext` wraps the `Page` component so It will make unnecessary requests, it won't do any harm to make the request apart from some `404`s responses.